File: code_challenge_solution/java/201 To-Do List Application.java
Sure! Here's the Java code to create a To-Do List Application:

```java
import java.util.ArrayList;
import java.util.Scanner;

public class ToDoListApp {
    public static void main(String[] args) {
        ArrayList<String> toDoList = new ArrayList<>();

        Scanner scanner = new Scanner(System.in);

        int choice = 0;

        while (choice != 4) {
            displayMenu();
            choice = scanner.nextInt();

            switch (choice) {
                case 1:
                    addItem(toDoList, scanner);
                    break;
                case 2:
                    removeItem(toDoList, scanner);
                    break;
                case 3:
                    displayToDoList(toDoList);
                    break;
                case 4:
                    System.out.println("Exiting the application...");
                    break;
                default:
                    System.out.println("Invalid choice! Please try again.");
                    break;
            }
        }

        scanner.close();
    }

    private static void displayMenu() {
        System.out.println("\nTo-Do List Application");
        System.out.println("======================");
        System.out.println("1. Add Item");
        System.out.println("2. Remove Item");
        System.out.println("3. Display To-Do List");
        System.out.println("4. Exit");
        System.out.print("Enter your choice: ");
    }

    private static void addItem(ArrayList<String> toDoList, Scanner scanner) {
        System.out.print("Enter the item to add: ");
        String item = scanner.next();
        toDoList.add(item);
        System.out.println("Item added successfully!");
    }

    private static void removeItem(ArrayList<String> toDoList, Scanner scanner) {
        System.out.print("Enter the item to remove: ");
        String item = scanner.next();
        if (toDoList.contains(item)) {
            toDoList.remove(item);
            System.out.println("Item removed successfully!");
        } else {
            System.out.println("Item not found in the list.");
        }
    }

    private static void displayToDoList(ArrayList<String> toDoList) {
        System.out.println("To-Do List:");
        if (toDoList.isEmpty()) {
            System.out.println("No items in the list.");
        } else {
            for (String item : toDoList) {
                System.out.println("- " + item);
            }
        }
    }
}
```

This code creates a To-Do List Application that allows the user to add, remove, and display items in a to-do list. It uses an `ArrayList` to store the items and a `Scanner` to read user input. The menu options are displayed in a loop until the user chooses to exit the application.

File: code_challenge_solution/c++/201 To-Do List Application.cpp
```c++
#include <iostream>
#include <vector>
#include <string>

using namespace std;

void printMenu()
{
    cout << "To-Do List Application" << endl;
    cout << "---------------------" << endl;
    cout << "1. Add Item" << endl;
    cout << "2. Delete Item" << endl;
    cout << "3. View List" << endl;
    cout << "4. Exit" << endl;
    cout << "---------------------" << endl;
}

void addItem(vector<string>& list)
{
    cout << "Enter the item to add: ";
    string item;
    cin.ignore();
    getline(cin, item);
    list.push_back(item);
    cout << "Item added successfully!" << endl;
}

void deleteItem(vector<string>& list)
{
    if (list.empty())
    {
        cout << "The list is empty. No items to delete." << endl;
    }
    else
    {
        cout << "Enter the index of the item to delete (0-" << list.size() - 1 << "): ";
        int index;
        cin >> index;

        if (index < 0 || index >= list.size())
        {
            cout << "Invalid index. Please try again." << endl;
        }
        else
        {
            list.erase(list.begin() + index);
            cout << "Item deleted successfully!" << endl;
        }
    }
}

void viewList(const vector<string>& list)
{
    if (list.empty())
    {
        cout << "The list is empty." << endl;
    }
    else
    {
        cout << "To-Do List:" << endl;
        for (int i = 0; i < list.size(); i++)
        {
            cout << i << ". " << list[i] << endl;
        }
    }
}

int main()
{
    vector<string> todoList;
    int choice = 0;

    while (choice != 4)
    {
        printMenu();
        cout << "Enter your choice (1-4): ";
        cin >> choice;

        switch (choice)
        {
            case 1:
                addItem(todoList);
                break;
            case 2:
                deleteItem(todoList);
                break;
            case 3:
                viewList(todoList);
                break;
            case 4:
                cout << "Exiting the application. Goodbye!" << endl;
                break;
            default:
                cout << "Invalid choice. Please try again." << endl;
        }

        cout << endl;
    }

    return 0;
}
```